packages/tracing/src/span.ts

```diff
@@ -334,6 +350,24 @@ export class Span implements SpanInterface {
       trace_id: this.traceId,
     });
   }
+
+  /**
+   *
+   * @param baggage
+   * @returns
+   */
+  private _getBaggageWithSentryValues(baggage: Baggage = createBaggage({})): Baggage {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+    const hub: Hub = ((this.transaction as any) && (this.transaction as any)._hub) || getCurrentHub();
```

Comment:
To me this seems a bit dangerous. I was wondering if we can get away with only having `getCurrentHub()` here, because in a node context, we'll get the right hub anyways because of domains, and in a browser context we only have one proper hub anyways (?).

I'll let you make the final call on this though. It's probably a bit dangerous but should also be fine.